packages/analytics/amplify_analytics_pinpoint/lib/src/analytics_plugin_impl.dart

```diff
@@ -21,16 +21,23 @@ import 'package:amplify_secure_storage/amplify_secure_storage.dart';
 
 /// {@template amplify_analytics_pinpoint.analytics_plugin_impl}
 /// The AWS Pinpoint implementation of the Amplify Analytics category.
+///
+/// To change the default behavior of global property storage,
+/// provide a [keyValueStore] value. If no value is provided,
+/// [AmplifySecureStorage] will be used with a `scope` of "analyticsPinpoint".
+///
 /// {@endtemplate}
 class AmplifyAnalyticsPinpoint extends AmplifyAnalyticsPinpointDart {
   /// {@macro amplify_analytics_pinpoint.analytics_plugin_impl}
-  AmplifyAnalyticsPinpoint()
-      : super(
-          keyValueStore: AmplifySecureStorage(
-            config: AmplifySecureStorageConfig(
-              scope: 'analyticsPinpoint',
-            ),
-          ),
+  AmplifyAnalyticsPinpoint({
```

Comment:
@fjnoyp @Jordan-Nelson thoughts on this? This was needed for my testing use case but wasn't sure if others would want an escape hatch like auth. The alternative would be bringing in the dart plugin in the integ test and overriding that way I suppose.

Reply:
Can you make the following changes:
- Make this `@visibleForTesting`. Exposing this for overriding within an app probably deserves more discussion. Allowing it to be overridden for testing seems reasonable though.
- add a TODO to rename this. `keyValueStore` doesn't describe this well. Probably want a better name if it will be exposed (even if just for testing).


```dart
// TODO(fjnoyp): Rename to reflect the data that is stored.
@visibleForTesting SecureStorageInterface? keyValueStore,
```

Reply:
just made this change, thanks, leaving thread open for Kyle to comment

Reply:
Yeah had the same problem when developing, the annotation visible for testing looks good.  Thanks for keeping open for me to see.